### DISCUSSION TITLE
Seeking Advice: Is a 7-Day Challenge Period Necessary for Op-stack L2 Network Setup?

### DISCUSSION THREAD

**Discussion by smallyunet** [ethereum-optimism/developers#484] [Application Development] [answered]
I'm currently using op-stack to set up an L2 network, and I was wondering if I need to set a 7-day challenge period?

I noticed in the documentation that the current op-stack hasn't released the fault proofs feature yet. From my understanding, I'll be running 4 types of nodes:

- op-geth
- op-node
- op-batcher
- op-proposer
Of these, I operate the op-batcher and op-proposer myself, and there can only be one node for each.

When deploying L1 smart contracts, there's a parameter in the configuration file for setting the challenge period duration. Would it be necessary for me to set it to 7 days? Since the proposer is operated by me, there's no risk of fraud. Could I possibly set the challenge period to 120 seconds?

Additionally, I'd like to kindly ask if it's possible for my users to run a larger number of op-geth and op-node nodes? Is it safe to do so?

**Comment by ghost** [comment, marked as answer]
@smallyunet If you are interested, I have written an alternative solution for op-stack chains that could possibly replace the current security model https://github.com/ethereum-optimism/optimism/issues/8037
